projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -1140,14 +1140,25 @@ function ( $category_id ) {
 					break;
 
 				case 'in_site_migration_flow':
```

Comment:
I think the whitelist is a good idea. It should have two values though, `site-migration` and `migration-signup`.

We also don't have to treat `$value` as a boolean anymore. I think Calypso should just send a string value with the flow name. If the string value is empty, we delete the option.

So the `switch` statement would be something like this (untested):
```
case 'in_site_migration_flow':
  if ( empty( $value ) ) {
    delete_option( 'in_site_migration_flow' );
    break;
  }

  $migration_flow_whitelist = array(
    'site-migration',
    'migration-signup',
  );

  if ( in_array( $value, $migration_flow_whitelist, true ) ) {
    update_option( 'in_site_migration_flow', $value );
  }

  $updated[ $key ] = $value;
  break;
```

If the string value is not empty and in the whitelist, then we update the option.

Reply:
Hum... that makes sense. However, the problem is synchronizing the deployment of this change. We should add a clean-up after we deploy this change and the front end; otherwise, some people may get stuck in a weird state.

Reply:
Just noting our conversation in Slack! I don't think we have to worry about cleanup because the `in_site_migration_flow` setting isn't being used yet. I think we can go ahead and make this change without it affecting previous migration users.